internal/verifier/check.go

```diff
@@ -166,19 +170,30 @@ func (verifier *Verifier) CheckDriver(ctx context.Context, filter map[string]any
 		verifier.phase = Idle
 	}()
 
-	verifier.mux.RLock()
-	csRunning := verifier.changeStreamRunning
-	verifier.mux.RUnlock()
-	if csRunning {
-		verifier.logger.Debug().Msg("Check: Change stream already running.")
+	if verifier.srcChangeStreamReader.changeStreamRunning {
+		verifier.logger.Debug().Msg("Check: Source change stream already running.")
 	} else {
-		verifier.logger.Debug().Msg("Change stream not running; starting change stream")
+		verifier.logger.Debug().Msg("Source change stream not running; starting change stream")
 
-		err = verifier.StartChangeStream(ctx)
+		err = verifier.srcChangeStreamReader.StartChangeStream(ctx)
 		if err != nil {
 			return errors.Wrap(err, "failed to start change stream on source")
 		}
+		verifier.StartChangeEventHandler(ctx, verifier.srcChangeStreamReader)
+	}
+
+	if verifier.dstChangeStreamReader.changeStreamRunning {
+		verifier.logger.Debug().Msg("Check: Destination change stream already running.")
+	} else {
+		verifier.logger.Debug().Msg("Destination change stream not running; starting change stream")
+
+		err = verifier.dstChangeStreamReader.StartChangeStream(ctx)
+		if err != nil {
+			return errors.Wrap(err, "failed to start change stream on destination")
+		}
+		verifier.StartChangeEventHandler(ctx, verifier.dstChangeStreamReader)
```

Comment:
Since ChangeStreamReader has a String() method, you could just loop over the src & dst change streams rather than duplicating this logic.